devlib/utils/ssh.py

```diff
@@ -299,6 +312,469 @@ def __init__(self,
         self.conn = ssh_get_shell(host, username, password, None, port, timeout, True, original_prompt)
 
 
+class Gem5Connection(TelnetConnection):
+
+    def __init__(self,
+                 platform,
+                 host=None,
+                 username=None,
+                 password=None,
+                 port=None,
+                 timeout=None,
+                 password_prompt=None,
+                 original_prompt=None,
+                 ):
+        # TODO: need to insert check that host is the actual place
+        # where devlib is now running
+        self.host = host
+        if username is not None or username is not 'root':
+            gem5_logger.info("User is always root in gem5")
+        if password is not None or password is not '':
+            gem5_logger.info("No password needed in gem5")
```

Comment:
This should be changed to raise a config error if they are not valid rather than set debugging message.